lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -365,6 +365,32 @@ struct AShrOpConversion : public OpConversionPattern<AShrOp> {
   }
 };
 
+struct ProcedureOpConversion : public OpConversionPattern<ProcedureOp> {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(ProcedureOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    switch (adaptor.getKind()) {
+    case ProcedureKind::AlwaysComb:
+    case ProcedureKind::Always:
+    case ProcedureKind::AlwaysFF:
+    case ProcedureKind::AlwaysLatch:
+    case ProcedureKind::Final:
+      return emitError(op->getLoc(), "Unsupported procedure operation");
+      return failure();
+    case ProcedureKind::Initial:
+      rewriter.setInsertionPointToEnd(
+          op->getParentOfType<mlir::ModuleOp>()->getBlock());
+      rewriter.replaceOpWithNewOp<llhd::ProcOp>(
+          op,
+          mlir::FunctionType::get(rewriter.getContext(), std::nullopt,
+                                  std::nullopt),
+          uint64_t(0), mlir::ArrayAttr(), mlir::ArrayAttr());
```

Comment:
I just realized that this just creates the `llhd.proc` op, but the body block and region of the `moore.procedure` is never moved over tho the `llhd.proc`. So this essentially creates a new empty `llhd.proc` and deletes the old `moore.procedure`, including its contents. I think you need to detach the `moore.procedure`'s body block and attach it to `llhd.proc` to have it carry over 😃

Reply:
Another one is about inserting to the end. I also got a crash but have no usefule message from stackflow information. So I make it annotation.
```
     // rewriter.setInsertionPointToEnd(
      //   op->getParentOfType<mlir::ModuleOp>()->getBlock());
```